### PR TITLE
Manager

### PR DESCRIPTION
**Pull request by EItanya**

### CODE REVIEW

File: pkg/manager/gomod.go [hunk 1]
```diff
@@ -0,0 +1,338 @@
...
+	zglob "github.com/mattn/go-zglob"
+	"github.com/rotisserie/eris"
+	"github.com/solo-io/protodep/pkg/modutils"
+	"github.com/solo-io/protodep/protodep"
+	"github.com/spf13/afero"
+)
+
+const (
+	ProtoMatchPattern   = "**/*.proto"
+	SoloKitMatchPattern = "**/solo-kit.json"
```
Comment (ilackarms) [inline pkg/manager/gomod.go]: hard couple to solo kit?

File: pkg/manager/common.go [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+	"io"
+	"log"
+	"os"
+	"path/filepath"
+
+	"github.com/rotisserie/eris"
+	"github.com/spf13/afero"
+)
+
+type FileCopier interface {
```
Comment (ilackarms) [inline pkg/manager/common.go:22]: comment: what is a FileCopier and how is it used?

File: pkg/manager/gomod.go [hunk 3]
```diff
@@ -0,0 +1,338 @@
...
+				GoMod: SoloKitProtoMatcher,
+			},
+		},
+		{
+			ImportType: &protodep.Import_GoMod{
+				GoMod: GogoProtoMatcher,
+			},
+		},
+	}
+)
```
Comment (ilackarms) [inline pkg/manager/gomod.go:19]: what is the reasoning behind hard-coding dependencies on the projects that will import this? these are circular dependencies (and seems like the wrong place to define them)

Comment (EItanya) [reply pkg/manager/gomod.go:19]: Leftover from solo-kit work, will remove

File: pkg/manager/gomod.go (hunk 1 above)
Comment (EItanya) [reply pkg/manager/gomod.go]: left-over from solo-kit will remove

File: protodep/defaults.go [hunk 4]
```diff
@@ -7,5 +7,7 @@ const (
 	// default directory into which proto, and other files will be vendored.
 	// Originally this was meant to be the vendor directory, but clashes with the go vendor directory
 	// meant it would be easier for this to inhabit it's own folder
-	DefaultDepDir = ".proto"
+	// See this section for more info: https://tip.golang.org/doc/go1.14#go-command
+	// This tool should not force to projects to build using vendor.
+	DefaultDepDir = ".proto_vendor"
```
Comment (ilackarms) [inline protodep/defaults.go]: nit: rename to `.anyvendor` (or whatever we rename this project to)

File: pkg/manager/protodep.go [hunk 5]
```diff
@@ -0,0 +1,42 @@
+package manager
+
+import (
+	"context"
+
+	"github.com/solo-io/protodep/protodep"
+)
+
+type DepFactory interface {
```
Comment (ilackarms) [inline pkg/manager/protodep.go]: what is a `DepFactory` and how is it used?

File: pkg/manager/protodep.go [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+	Ensure(ctx context.Context, opts *protodep.Config) error
+}
+
+type Manager struct {
```
Comment (ilackarms) [inline pkg/manager/protodep.go:21]: what is a `Manager` and how is it used?

File: pkg/manager/protodep.go [hunk 7]
```diff
@@ -0,0 +1,42 @@
+package manager
+
+import (
+	"context"
+
+	"github.com/solo-io/protodep/protodep"
+)
+
+type DepFactory interface {
+	Ensure(ctx context.Context, opts *protodep.Config) error
```
Comment (ilackarms) [inline pkg/manager/protodep.go]: nit: `depFactory.Ensure` doesn't seem like a clear API to me. how does this relate to how `protodep` is used? i assume this is the API called by the user?

File: ci/check-code-gen.sh [hunk 8]
```diff
@@ -9,11 +9,8 @@ if [ ! -f .gitignore ]; then
 fi
 
 
-git init
-git config user.email "you@example.com"
+git config --global user.email "you@example.com"
```
Comment (yuval-k) [inline ci/check-code-gen.sh]: changing global git config is a bit scary here.
i think we can remove this as we are not doing a git commit.
if we are doing a git commit, can we add the author using the `--author "Your Name <you@example.com>"` flag in the git commit command?

Comment (EItanya) [reply ci/check-code-gen.sh]: deleted as we're no longer doing a commit in this step

File: pkg/manager/common.go [hunk 9]
```diff
@@ -0,0 +1,122 @@
...
+	if goPath == "" {
+		// the default GOPATH for go v1.11
+		goPath = filepath.Join(os.Getenv("HOME"), "go")
+	}
+
+	var normPath string
+
+	// go mod replaces capital letters with "!" and then the lower case.
+	// This checks for that and switches it so we can find the file
+	for _, char := range importPath {
```
Comment (yuval-k) [inline pkg/manager/common.go:66]: should this be a copy-pate of this?
https://github.com/golang/go/blob/c54bc3448390d4ae4495d6d2c03c9dd4111b08f1/src/cmd/go/internal/module/module.go#L455

**Review by ilackarms — APPROVED** [review]

File: pkg/modutils/mirror.go [hunk 10]
```diff
@@ -0,0 +1,24 @@
+package modutils
+
+import (
+	"time"
+)
+
+type Module struct {
```
Comment (yuval-k) [inline pkg/modutils/mirror.go:9]: this seems to only be used in tests.
can we add a test build tag to guarantee that?

**Pull request merged**